### DISCUSSION TITLE
AppCheck already-activated error on Flutter hot reload / hot restart

### DISCUSSION THREAD

**Discussion by misteroak** [firebase/flutterfire#8017] [Q&A]
Hi,

I am likely missing something here, but couldn't find an answer online. I've been following the instructions to add AppCheck to my Flutter Web app. It's working fine on first load, but the moment I hit hot reload or hot restart (using Vs Code) I get the following error:

`Error: [app-check/already-activated] AppCheck: You are trying to activate AppCheck for FirebaseApp [DEFAULT], while it is already activated. AppCheck can only be activated once. .
`

Here's my initialization code:

```
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );

  await FirebaseAppCheck.instance.activate(
    webRecaptchaSiteKey: READCTED-TOKEN,
  );

  BlocOverrides.runZoned(
    () {
      runApp(MyApp());
    },
    blocObserver: AppBlocObserver(),
  );
}
```

What am I doing wrong?

**Comment by gavinerasmus** [comment]
Looks like an issue they are aware of and working on: https://github.com/FirebaseExtended/flutterfire/issues/8019 
 Is annoying... have to kill the debug and restart the app each time...
